yaksh/views.py

```diff
@@ -605,7 +606,7 @@ def course_detail(request, course_id):
     user = request.user
     if not is_moderator(user):
         raise Http404('You are not allowed to view this page')
-    course = get_object_or_404(Course, creator=user, pk=course_id)
+    course = Course.objects.filter(Q(creator=user)|Q(teachers=user), pk=course_id)[0]
```

Comment:
Is there a better way to express this? This will fail in case there is a user who is neither a Creator nor a Teacher related to that course.(indexing an empty list will throw IndexError)